src/Phinx/Db/Adapter/PdoAdapter.php

```diff
@@ -173,33 +173,105 @@ public function fetchAll($sql)
      * {@inheritdoc}
      */
     public function insert(Table $table, $row)
+    {
+        if ($this->isDryRunEnabled()) {
+            $this->dryRunInsert($table, $row);
+        } else {
+            $this->doInsert($table, $row);
+        }
+    }
+
+    /**
+     * Inserts data into the table.
+     *
+     * @param Table $table  The table where to insert the data
+     * @param array $row    The data to be inserted as assoc array column name -> value
+     * @return void
+     */
+    private function doInsert(Table $table, $row)
     {
         $sql = sprintf(
-            "INSERT INTO %s ",
+            'INSERT INTO %s ',
             $this->quoteTableName($table->getName())
         );
 
         $columns = array_keys($row);
-        $sql .= "(" . implode(', ', array_map([$this, 'quoteColumnName'], $columns)) . ")";
-        $sql .= " VALUES (" . implode(', ', array_fill(0, count($columns), '?')) . ")";
+        $sql .= '(' . implode(', ', array_map([$this, 'quoteColumnName'], $columns)) . ')';
+        $sql .= ' VALUES (' . implode(', ', array_fill(0, count($columns), '?')) . ')';
 
         $stmt = $this->getConnection()->prepare($sql);
         $stmt->execute(array_values($row));
     }
 
+    /**
+     * Dumps the insert SQL statement.
+     *
+     * @param Table $table  The table where to insert the data
+     * @param array $row    The data to be inserted as assoc array column name -> value
+     * @return void
+     */
+    private function dryRunInsert(Table $table, $row)
+    {
+        $sql = sprintf(
+            'INSERT INTO %s ',
+            $this->quoteTableName($table->getName())
+        );
+
+        $columns = array_keys($row);
+        $sql .= '(' . implode(', ', array_map([$this, 'quoteColumnName'], $columns)) . ')';
+        $sql .= ' VALUES (' . implode(', ', array_map([$this, 'quoteValue'], $row)) . ');';
```

Comment:
Please do not copy&paste. You can add a new private method and use it in `runInsert()` and `dryRunInsert()`. But please wait for a while before doing the work. I think we should discuss how to implement dry-run feature first.

Reply:
Sorry, I misread. It was not a full copy&pasted.